sass/assets.scss

```diff
@@ -1,3 +1,58 @@
+// Stores information about the assets that are available
+// The structure of the map is that the keys indicate the module
+// to which the assets belong. When the module is null, the assets
+// belong to the application itself. Application assets take priority
+// over module assets, that is, "foo/bar.png" in the application will
+// take priority over "bar.png" in the "foo" module.
+// The values of this map is another map of source urls to the full
+// path on disk where the asset resides.
+$eg-registered-assets: () !default;
+
+// Registers many assets for a module at once.
+// @param $module string The name of the module to which these assets belong.
+// @param $assets map<string => string> maps source urls to the full path of the asset.
+@mixin register-asset($module, $url, $path, $uri: null) {
```

Comment:
For convention, should these functions all be `asset-*`? I know that'd make the `register-assets` a lot wordier, but not sure we want to be in multiple namespaces.